eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/DependencyAnalyzer.java

```diff
@@ -0,0 +1,207 @@
+
+                addDependencyToDependencyTree(parentDependency, null, nameToVersionToChildrenDependencyTree);
+
+                for (MavenArtifactInfo dependency : dependencies) {
+                    if (dependency.getScope() == ScopeType.TEST) {
+                        continue;
+                    }
+                    if(RESOLVED_EXCLUSION_LIST.contains(dependency.getCoordinate().getArtifactId())) {
+                        // This is coming from the BOM, we may not include this.
+                        continue;
+                    }
+
+                    BomDependency childDependency = new BomDependency(dependency.getCoordinate());
+                    addDependencyToDependencyTree(childDependency, parentDependency, nameToVersionToChildrenDependencyTree);
+                }
+            }
+            catch(Exception ex) {
+                System.out.println(ex);
+            }
+        }
+    }
+
+    private static MavenResolverSystemBase<PomEquippedResolveStage, PomlessResolveStage, MavenStrategyStage, MavenFormatStage> getMavenResolver() {
+        return Maven.configureResolver().withMavenCentralRepo(true);
+    }
+
+    private static void addDependencyToDependencyTree(BomDependency dependency, BomDependency parentDependency, TreeMap<BomDependencyNoVersion, HashMap<String, Collection<BomDependency>>> dependencyTree) {
+        if (!dependencyTree.containsKey(dependency)) {
+            dependencyTree.put(new BomDependencyNoVersion(dependency.getGroupId(), dependency.getArtifactId()), new HashMap<>());
+        }
+
+        HashMap<String, Collection<BomDependency>> versionToParents = dependencyTree.get(dependency);
+        if(!versionToParents.containsKey(dependency.getVersion())) {
+            versionToParents.put(dependency.getVersion(), new ArrayList<>());
+        }
+
+        if(parentDependency != null) {
+            versionToParents.get(dependency.getVersion()).add(parentDependency);
+        }
+    }
+
+
+    private void makeDependencyInEligible(BomDependency dependency) {
+        if(nameToVersionToChildrenDependencyTree.containsKey(dependency)) {
+            HashMap<String, Collection<BomDependency>> versionToDependency = nameToVersionToChildrenDependencyTree.get(dependency);
+            bomIneligibleDependencies.add(dependency);
+
+            // Make all the dependencies that include these also ineligible.
+            versionToDependency.get(dependency.getVersion()).forEach(parent -> makeDependencyInEligible(parent));
+        }
+    }
+
+    private boolean resolveConflicts() {
+        AtomicBoolean hasConflict = new AtomicBoolean(false);
+        // TODO: Consider having a non-core plan of picking the highest version with the maximum dependant children.
+        nameToVersionToChildrenDependencyTree.keySet().stream().forEach(
+            key -> {
+                HashMap<String, Collection<BomDependency>> versionToDependency = nameToVersionToChildrenDependencyTree.get(key);
```

Comment:
It's better to always code against an interface instead of an implementation. Use `Map` instead of `HashMap` here.